build_library/build_image_util.sh

```diff
@@ -273,6 +273,58 @@ write_packages() {
     image_packages "$1" | sort > "$2"
 }
 
+get_metadata() {
+    local prefix="$1"
+    local pkg="$2"
+    local key="$3"
+    local path="${prefix}/var/db/pkg/${pkg%%:*}/${key}"
+    local val
+    if [[ -f "$path" ]]; then
+        val="$(< $path)"
+    else
+        # The package is not installed in $prefix or $key not exposed as file,
+        # so get the value from its ebuild
+        val=$(portageq-${BOARD} metadata "${BOARD_ROOT}" ebuild \
+                    "${pkg%%:*}" "${key}" 2>/dev/null ||:)
+    fi
+    if [ "${key}" = SRC_URI ]; then
```

Comment:
Can you please add a comment about this conditional? i.e. Why is it needed, what it's doing, etc. It's pretty long, so I think it's worth explaining it a bit more.

Also, not that it matters, but I think it would be clearer if SRC_URI was between quotes.

Reply:
Done